File: src/sfqc-jni/src/SFQueueClientInterface.java
package jp.co.iret.sfq;

public interface SFQueueClientInterface
{
	public String push(java.util.Map<String, Object> params) throws SFQueueClientException;
	public java.util.Map<String, Object> pop() throws SFQueueClientException;
	public java.util.Map<String, Object> shift() throws SFQueueClientException;
}

File: src/sfq-cs/sfq/SFQueueClientInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sfq
{
    using ObjectMap = Dictionary<String, Object>;

    public interface SFQueueClientInterface
    {
        String push(ObjectMap param);
        ObjectMap pop(ObjectMap param);
        ObjectMap shift(ObjectMap param);
        ObjectMap pop();
        ObjectMap shift();
    }
}
